File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/pagers/ItemAwareFragmentPagerAdapter.java
package com.exact.twitch.ui.pagers;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import android.view.ViewGroup;

public abstract class ItemAwareFragmentPagerAdapter extends FragmentPagerAdapter {

    private Fragment currentFragment;

    ItemAwareFragmentPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    Fragment getCurrentFragment() {
        return currentFragment;
    }

    @Override
    public void setPrimaryItem(ViewGroup container, int position, Object object) {
        super.setPrimaryItem(container, position, object);
        currentFragment = (Fragment) object;
    }
}

File: app/src/main/java/com/exact/twitch/ui/pagers/ItemAwareFragmentPagerAdapter.kt
package com.exact.twitch.ui.pagers

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import android.view.ViewGroup

abstract class ItemAwareFragmentPagerAdapter internal constructor(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    internal lateinit var currentFragment: Fragment
        private set

    override fun setPrimaryItem(container: ViewGroup, position: Int, `object`: Any) {
        super.setPrimaryItem(container, position, `object`)
        currentFragment = `object` as Fragment
    }
}
